### ISSUE TITLE
Feature request: add an option to disable alerts log rotation

### ISSUE THREAD

**Issue by jchenrev** [wazuh/wazuh#5618]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
|3.11.4|analysisd|Manager|Package/Puppet|Ubuntu 18.04|

The current behavior of my Wazuh manager is rotating logs every day and never deletes them. For example

```
$ ll /var/ossec/logs/alerts/2020/Jul/ | head
total 365008
drwxr-x--- 2 ossec ossec     4096 Aug  1 00:00 ./
drwxr-x--- 5 ossec ossec     4096 Aug  1 00:00 ../
-rw-r----- 1 ossec ossec  7949432 Jul  2 00:00 ossec-alerts-01.json.gz
-rw-r----- 1 ossec ossec      631 Jul  2 00:00 ossec-alerts-01.json.sum
-rw-r----- 1 ossec ossec  8425761 Jul  2 00:00 ossec-alerts-01.log.gz
-rw-r----- 1 ossec ossec      628 Jul  2 00:00 ossec-alerts-01.log.sum
-rw-r----- 1 ossec ossec  7171384 Jul  3 00:00 ossec-alerts-02.json.gz
-rw-r----- 1 ossec ossec      631 Jul  3 00:00 ossec-alerts-02.json.sum
-rw-r----- 1 ossec ossec  7256190 Jul  3 00:00 ossec-alerts-02.log.gz
```

My server has relatively small disk space so I want to do size based rotation instead of just leaving the logs forever.
I've checked out other tickets and it seems like the size based rotation is not implemented and won't be implemented soon.
So I'd like to just disable wazuh's built-in log rotation and use an external tool like logrotate. 

If it helps, my current logging configuration is like

```
$ curl -u foo:bar localhost/agents/000/config/analysis/global?pretty
{
   "error": 0,
   "data": {
      "global": {
         "email_notification": "yes",
         "logall": "no",
         "logall_json": "no",
         "integrity_checking": 8,
         "rootkit_detection": 8,
         "host_information": 8,
         "prelude_output": "no",
         "zeromq_output": "no",
         "jsonout_output": "yes",
         "alerts_log": "yes",
         "stats": 4,
         "memory_size": 8192,
         "white_list": [
            "127.0.0.1",
            "10.0.0.2",
            "localhost.localdomain"
         ],
         "rotate_interval": 0,
         "max_output_size": 0
      }
   }
}
```
```
$ curl -u foo:bar localhost/agents/000/config/analysis/internal?pretty
{
   "error": 0,
   "data": {
      "internal": {
         "analysisd": {
            "debug": 0,
            "default_timeframe": 360,
            "stats_maxdiff": 999000,
            "stats_mindiff": 1250,
            "stats_percent_diff": 150,
            "fts_list_size": 32,
            "fts_min_size_for_str": 14,
            "log_fw": 1,
            "decoder_order_size": 256,
            "label_cache_maxage": 1,
            "show_hidden_labels": 0,
            "rlimit_nofile": 65536,
            "min_rotate_interval": 600
         }
      }
   }
}
```

**Comment by 0xN3utr0n** [comment]
Hi @jchenrev,
If it's only for `alerts.log/json` logs, the module [monitord](https://documentation.wazuh.com/3.13/user-manual/reference/daemons/ossec-monitord.html) does exactly what you want. For instance, among the available [options](https://documentation.wazuh.com/3.13/user-manual/reference/internal-options.html#monitord), you can specify:
- The maximum size per log:
`monitord.size_rotate`
- Or to turn off rotation:
`monitord.rotate_log`

As for the deletion, you can just run a `cron job` periodically that checks whether the existent log files meet your requirements or not, and consequently remove them.

Hope it helps,
Fernando

**Comment by jchenrev** [comment]
@0xN3utr0n Thanks for you quick response. As mentioned in #2502, I thought `monitord.rotate_log` is only for ossec.log?
Can you explain it a bit more because there's no documentation of monitord except the one you point me to, which only contains some simple description. 

It has an option `monitord.keep_log_days`, which is set to 31. But my alerts logs are definitely kept more than 31 days.

`monitord.daily_rotations` "Maximum number of rotations per day for internal logs." What does it mean? Isn't that the log will only be rotated once per day?

Here is the monitord config in internal_options.conf
```
# Monitord day_wait. Amount of seconds to wait before rotating/compressing/signing [0..600]
# the files.
monitord.day_wait=10

# Monitord compress. (0=do not compress, 1=compress)
monitord.compress=1

# Monitord sign. (0=do not sign, 1=sign)
monitord.sign=1

# Monitord monitor_agents. (0=do not monitor, 1=monitor)
monitord.monitor_agents=1

# Rotate plain and JSON logs daily. (0=no, 1=yes)
monitord.rotate_log=1

# Days to keep old ossec.log files [0..500]
monitord.keep_log_days=31

# Size of internal log files to rotate them (Megabytes) [0..4096]
monitord.size_rotate=512

# Maximum number of rotations per day for internal logs [1..256]
monitord.daily_rotations=12

# Number of minutes for deleting an disconnected agent [0..120]. (0=disabled)
monitord.delete_old_agents=0
```

**Comment by 0xN3utr0n** [comment]
@jchenrev My bad, it looks like `analysisd` is the one in charge over the `alerts.log/json` files rather than `monitord`. 

Good news is that it's still possible to do it. Have a look at #318, it has a detailed explanation about how to correctly enable such features. 

**tl;dr**

- `<rotate_interval>0</rotate_interval>` Will disable the log rotation for the alerts logs.

- `<max_output_size>20M</max_output_size>` Will set the size limit as per your request. (`rotate_interval` must be enabled)

I've tested this myself, so I can assure it works fine.

Kind regards,
Fernando

**Comment by jchenrev** [comment]
@0xN3utr0n Thanks. 
But I'm still a bit confused because the docs says the default `rotate_interval` is 0 and I didn't find it in my config so it's using the default but it's still rotating the logs.
Anyway I will try it out.

**Comment by 0xN3utr0n** [comment]
Hi @jchenrev , 
After further investigation (#3072), it seems that the proposed solution doesn't work as expected, so no `disable` switch is available right now. That being said, `max_output_size` works perfectly fine, as noted in my previous message, if you want to use it instead of `logrotate`.

Improving log rotation for the alerts is in our [roadmap](https://github.com/wazuh/wazuh/issues/3072#issuecomment-531675715), so sooner rather than later we will include that feature.

Kind regards,
Fernando

**Comment by od55** [comment]
no any ipdates for disabling?
